docs/developers/content-creation/creating-a-role.md

```diff
@@ -40,3 +44,526 @@ suicideMultiplier = -1
+
+`ttt2-role_test/lua/terrortown/lang/de/tester.lua`
+
+This would also add a german translation. Or:
+
+`ttt2-role_test/lua/terrortown/lang/it/tester.lua`
+
+For an italian translation. 
+
+A `[your role name here].lua` is then added to each language folder. We will discuss what this file does and what is written in it in a moment.
+
+For further information, go [this way](https://docs.ttt2.neoxult.de/developers/content-creation/language-support/#adding-a-language-file)
+
+#### The main files
+
+Of course, all files described here are important for an error-free operation of the role. Therefore, "Main Files" is perhaps even the wrong name. This refers to the files that contain the function code that gives the role the functionality it is supposed to fulfil. 
+That structure must look like this:
+
+`ttt2-role_test/lua/terrortown/entities/roles/tester/shared.lua`
+
+The interesting part here is the “tester”-folder and the `shared.lua`. This folder always bears the name of the role. The shared.lua is then the source file of the role. 
+
+You might ask, why this file is named `shared` and not `tester` like all the other files. 
+In Garry’s Mod we can code on 2 levels: The server and the client. The server handles the data that is transferred between the clients. Programming on server level means that we can change data that is affecting all clients or a certain group of clients etc. To be specific: Everything that has to do with the game mode or must be handled globally, and this includes the function of a role, is developed server-side. On the client’s side we can change things that is only by the client themselves, e.g. HUD elements. This also means that you cannot change anything related to TTT2 client-side. 
+
+In Garry’s Mod there are 2 ways to access either the client or the server:
+You can name a folder `server` or `client`. Every file in that folder then can be handled server-side or client-side. The problem: This is static. You cannot switch this while running the code. So these files will then always be handled only on the client or on the server.
+
+The 2. way now also explains, why the main code file is named `shared.lua`. If a file or folder is declared as shared, both client-side code and server-side code can be executed within it. And because we want to run both server and client code in the role code, the main file is called `shared.lua`.
+
+#### The ConVar files
+
+ConVars `(abbr. for “console variable”)` are used to store information on the server and the client by using the console. In TTT2 role coding we use them to give players the posibility change the options of the roles to their liking and thus customise each role to their playing style.
+
+The ConVar file must be placed here:
```

Comment:
no, it doesn't have to be there. Maybe say it like "It is recommended to put..."